### PR TITLE
Bugfix/issue 4684

### LINKED ISSUE #4684: Static root plugin not loading after namespace/key changes in OpenMCT
#### Summary
What it says on the tin! The static root plugin used to use a `namespace:key` format in the key field, which used to work in OpenMCT, but recent changes have made this format incompatible.

#### Impact Check List
- [X] Data loss or misrepresented data?
- [X] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?

#### Steps to Reproduce/Testing Steps
1. Install the static root plugin in OpenMCT using a JSON file with an object in it.
2. The plugin currently shows "Missing: root:1" in the tree for the first object. This should properly load and show the object defined in the file.

### PR DESCRIPTION
**Pull request by khalidadil**
Closes #4684 

### Describe your changes:
This code loops through the exported JSON while importing it to rewrite identifiers and update namespace/compositions/locations.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate unit tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/plugins/staticRootPlugin/StaticModelProvider.js [hunk 1]
```diff
@@ -9,30 +9,89 @@ define([
...
-                    key: i
+                objectLeaf[nodeKey] = parseTreeLeaf(nodeKey, objectLeaf[nodeKey], idMap, namespace);
+            }
+        });
+
+        return objectLeaf;
+    }
+
+    function parseArrayLeaf(arrayLeaf, idMap, namespace) {
+        arrayLeaf.forEach((leafValue, index) => {
```
Comment (nikhilmandlik) [inline src/plugins/staticRootPlugin/StaticModelProvider.js:31]: do we need to mutate `arrayLeaf`? if not can use `return arrayLeaf.map(.....`

File: src/plugins/staticRootPlugin/StaticModelProvider.js [hunk 2]
```diff
@@ -9,30 +9,89 @@ define([
...
+        Object.keys(objectTree).forEach((originalId, index) => {
+            if (originalId === importData.rootId) {
+                idMap.set(originalId, rootIdentifier.key);
+            } else {
+                idMap.set(originalId, index.toString());
             }
         });
 
-        return JSON.parse(objectString);
+        const newTree = parseTreeLeaf(null, objectTree, idMap, namespace);
```
Comment (nikhilmandlik) [inline src/plugins/staticRootPlugin/StaticModelProvider.js:92]: can  just `return parseTreeLeaf(null, objectTree, idMap, namespace);`

File: src/plugins/staticRootPlugin/StaticModelProvider.js [hunk 3]
```diff
@@ -9,30 +9,89 @@ define([
...
+    function rewriteObjectIdentifiers(importData, rootIdentifier) {
+        const namespace = rootIdentifier.namespace;
+        const idMap = new Map();
+        const objectTree = importData.openmct;
+
+        Object.keys(objectTree).forEach((originalId, index) => {
+            if (originalId === importData.rootId) {
+                idMap.set(originalId, rootIdentifier.key);
+            } else {
+                idMap.set(originalId, index.toString());
```
Comment (nikhilmandlik) [inline src/plugins/staticRootPlugin/StaticModelProvider.js:88]:
```
let newId = index.toString();
if (originalId === importData.rootId) {
   newId =  rootIdentifier.key;
} 
idMap.set(originalId, newId);
```

File: src/plugins/staticRootPlugin/StaticModelProvider.js [hunk 4]
```diff
@@ -9,30 +9,89 @@ define([
...
      */
-    function rewriteObjectIdentifiers(importData, rootIdentifier) {
-        const rootId = importData.rootId;
-        let objectString = JSON.stringify(importData.openmct);
-
-        Object.keys(importData.openmct).forEach(function (originalId, i) {
-            let newId;
-            if (originalId === rootId) {
-                newId = objectUtils.makeKeyString(rootIdentifier);
+    function parseObjectLeaf(objectLeaf, idMap, namespace) {
```
Comment (nikhilmandlik) [inline src/plugins/staticRootPlugin/StaticModelProvider.js:12]: Also lets convert it to ES6 modules.

**Review by nikhilmandlik — CHANGES_REQUESTED** [review]
small changes

File: src/plugins/staticRootPlugin/StaticModelProvider.js (hunk 2 above)
Comment (khalidadil) [reply src/plugins/staticRootPlugin/StaticModelProvider.js:92]: I like giving it a name here - it makes it more clear what's being returned

**Comment by khalidadil** [comment]
##Testing Instructions

To test this change:
1. Check out the `bugfix/issue-4684-testing` branch locally
2. Optional: Export some domainObjects from OpenMCT and rename to `static-root.json`. Replace the `static-root.json` file in the root directory of the local openmct repo.
3. Run `npm run build:dev` and `npm start`
4. Open the app and check that it loads correctly
5. Check that the Clock, Link, Web Page, and Display Layout objects load properly.

**Review by nikhilmandlik — APPROVED** [review]
LGTM

**Pull request merged**